source/utilities.cc

```diff
@@ -180,6 +181,23 @@ namespace aspect
     }
 
 
+    //Evaluate the cosine and sine terms of a real spherical harmonic.
+    //This is a fully normalized harmonic, that is to say, inner products
+    //of these functions should integrate to a kronecker delta over
+    //the surface of a sphere.
+    std::pair<double,double> real_spherical_harmonic( unsigned int l, //degree
+                                                      unsigned int m, //order
+                                                      double theta,   //colatitude (radians)
+                                                      double phi )    //longitude (radians)
+    {
+      const double sqrt_2 = numbers::SQRT2;
+      const std::complex<double> sph_harm_val = boost::math::spherical_harmonic( l, m, theta, phi );
+      if ( m == 0 )
+        return std::make_pair( sph_harm_val.real(), 0.0 );
+      else
+        return std::make_pair( sqrt_2 * sph_harm_val.real(), sqrt_2 * sph_harm_val.imag() );
```

Comment:
This is the sort of thing that ought to be documented in formulas. Maybe also mention in the documentation of the function that it uses the boost function, but with modifications. The current text sounds like it suggests that `boost::math::spherical_harmonic` is exactly what the function returns.